src/dmd/traits.d

```diff
@@ -93,56 +93,1048 @@ private Dsymbol getDsymbolWithoutExpCtx(RootObject oarg)
 
 extern (C++) __gshared StringTable traitsStringTable;
 
+/**
+If the arguments are all either types that are arithmetic types,
+or expressions that are typed as arithmetic types, then $(D true)
+is returned.
+Otherwise, $(D false) is returned.
+If there are no arguments, $(D false) is returned.
+
+---
+import std.stdio;
+
+void main()
+{
+    int i;
+    writeln(__traits(isArithmetic, int));
+    writeln(__traits(isArithmetic, i, i+1, int));
+    writeln(__traits(isArithmetic));
+    writeln(__traits(isArithmetic, int*));
+}
+---
+
+    $(P Prints:)
+
+$(CONSOLE
+true
+true
+false
+false
+)
+*/
+enum isArithmetic = "isArithmetic";
```

Comment:
I suggest using `__traits(identifier, isArithmetic)` to get the string. Instead of having to repeat all the enum names twice.

Reply:
Isn't that what .stringof is for? I see no problem really with the repetition, as the intent is clear.

Reply:
`.stringof` will result in a forward reference error.